common/hash_test.go

```diff
@@ -17,9 +17,19 @@
 package common
 
 import (
+	"bytes"
 	"testing"
 )
 
+func TestBlake2b218(t *testing.T) {
```

Comment:
Would probably be better to get samples from other implementations and test against those. You might also check Blake2b128 return size

Reply:
done